sycl/include/CL/sycl/detail/cg.hpp

```diff
@@ -89,6 +139,14 @@ class CG {
 
   CGTYPE getType() { return MType; }
 
+  std::shared_ptr<std::vector<ExtendedMemberT>> getExtendedMembers() {
+    if (getCGTypeVersion(MType) == static_cast<unsigned int>(CG_VERSION::V0) ||
+        MSharedPtrStorage.empty())
+      return nullptr;
+
+    return convertToExtendedMembers(MSharedPtrStorage[0]);
```

Comment:
Not clear why [0]. Can't CG contain other shared ptrs? Please clarify

Reply:
Tried to explain it in the comment on top of the file.

Reply:
Yes, the idea is to avoid adding a new field to cg and handler since adding them breaks ABI.